File: 20/day20p2.py
def get_present_count(N: int) -> int:
    result = 0
    i = 1
    while i * i <= N:
        if N % i == 0:
            first = i
            if first * 50 >= N:
                result += first * 11
            second = N // first
            if second != i and second * 50 >= N:
                result += second * 11
        i += 1

    return result

target = int(open('input2.txt').read().strip())
i = 1
while True:
    presents = get_present_count(i)
    if presents >= target:
        break
    i += 1

print(i)

File: 20/day20p2.cpp
#include <iostream>

int get_present_count(int n) {
    int result = 0;
    for(int i = 1; i * i <= n; ++i) {
        if(n % i == 0) {
            int first = i;
            if(first * 50 >= n) {
                result += first * 11;
            }
            int second = n / first;
            if(second != i && second * 50 >= n) {
                result += second * 11;
            }
        }
    }
    return result;
}

int main() {
    int target;
    std::cin >> target;
    int i = 1;
    while(true) {
        int presents = get_present_count(i);
        if(presents >= target) {
            break;
        }
        ++i;
    }
    std::cout << i << "\n";
}